staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,192 @@
+	return &SubresourceScaler{
+		parentEntries:      parentEntries,
+		parentGroupVersion: parentGroupVersion,
+		parentPath:         parentPath,
+	}, nil
+}
+
+// ToSubresource filter the managed fields of the main resource and convert
+// them so that they can be handled by a subresource.
+// For the managed fields that have a path that could be handled by a
+// subresource this method performs two changes:
+// 1. The APIVersion is changed to the APIVersion of the subresource
+// 2. The Fields are transformed to the path of that subresource
+func (h *SubresourceScaler) ToSubresource() ([]metav1.ManagedFieldsEntry, error) {
+	managed, err := internal.DecodeObjectManagedFields(h.parentEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	f := fieldpath.ManagedFields{}
+	t := map[string]*metav1.Time{}
+	for manager, versionedSet := range managed.Fields() {
+
+		if versionedSet.Set().Has(h.parentPath) {
+			newVersionedSet := fieldpath.NewVersionedSet(
+				fieldpath.NewSet(scalePath),
+				fieldpath.APIVersion(scaleGroupVersion.String()),
+				versionedSet.Applied(),
+			)
+
+			f[manager] = newVersionedSet
+			if operationTime, ok := managed.Times()[manager]; ok {
+				t[manager] = operationTime
+			}
+		}
+
+	}
+
+	return managedFieldsEntries(internal.NewManaged(f, t))
+}
+
+// ToParent merges the `subresourceEntries` with the parent entries and
+// transform them accordingly
+func (h *SubresourceScaler) ToParent(scaleEntries []metav1.ManagedFieldsEntry) ([]metav1.ManagedFieldsEntry, error) {
+	f := fieldpath.ManagedFields{}
+	t := map[string]*metav1.Time{}
+
+	// 1. Decode the scale entry and change it back to the parent API version and
+	// the parent path
+	scaleManagedFields, err := internal.DecodeObjectManagedFields(scaleEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	for manager, versionedSet := range scaleManagedFields.Fields() {
+
+		if versionedSet.Set().Has(scalePath) {
+			// TODO (nodo): handle multiple managed fields versions.
+			newAPIVersion := h.parentGroupVersion.String()
+			newVersionedSet := fieldpath.NewVersionedSet(
+				fieldpath.NewSet(h.parentPath),
+				fieldpath.APIVersion(newAPIVersion),
+				versionedSet.Applied(),
+			)
+			f[manager] = newVersionedSet
+			if operationTime, ok := scaleManagedFields.Times()[manager]; ok {
+				t[manager] = operationTime
+			}
+		}
+	}
+
+	// If the subresource entry does not have the scale path, then do nothing.
+	if len(f) == 0 {
+		return h.parentEntries, nil
+	}
+
+	// 2. Decode the parent fields and process them to merge them with the
+	// subresource.
+	parentManagedFields, err := internal.DecodeObjectManagedFields(h.parentEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	toRemove := fieldpath.NewSet(h.parentPath)
+	for manager, versionedSet := range parentManagedFields.Fields() {
+		// "steal" the scale path from the parent managed fields
+		newSet := versionedSet.Set().Difference(toRemove)
```

Comment:
Since you don't check the version of the `versionedSet`, you don't know if that field that you remove exist for that specific version?